crosshair/libimpl/builtinslib.py

```diff
@@ -1676,7 +1676,12 @@ def split(self, sep:Optional[str] = None, maxsplit:int = -1):
         new_maxsplit = -1 if maxsplit == -1 else maxsplit - 1
         ret.extend(self[first_occurance+1:].split(sep=sep, maxsplit=new_maxsplit))
         return ret
-
+    
+    def zfill(self, width: int):
+        size = SmtInt(z3.Length(self.var)) # Why does this need to be casted to SmtInt?
+        offset = width - size # Unsure about the type of offset at this point
```

Comment:
`size` has type `SmtInt`. `width` could be either a real `int` or another `SmtInt`. Either way, the subtraction operator is overridden and will produce an `SmtInt` as a result!